components/tenant-mgt/org.wso2.carbon.tenant.mgt.ui/src/main/java/org/wso2/carbon/tenant/mgt/ui/utils/TenantMgtUtil.java

```diff
@@ -121,9 +121,17 @@ public static void updateTenantConfigBean(HttpServletRequest request, ServletCon
             //update usage plan(subscription) per tenant
             //usagePlanClient.updateUsagePlan(tenantInfoBean);
         } catch (Exception e) {
-            String msg = "Failed to update the tenant config. tenant-domain: "
-                    + tenantInfoBean.getTenantDomain() + ", " + "tenant-admin: "
-                    + tenantInfoBean.getAdmin() + ".";
+            String msg = "Failed to update the tenant domain '" + tenantInfoBean.getTenantDomain() + "'.";
+
+            String rootCause = ((TenantMgtAdminServiceExceptionException) e).getFaultMessage().
+                    getTenantMgtAdminServiceException().getMessage();
+
+            if (rootCause != null && !rootCause.isEmpty()) {
+                msg = msg + " Root cause : " + rootCause;
```

Comment:
+1 for adding the root cause message to the error response since tenant addition is a super-admin task.